docs/compilers/CSharp/Compiler Breaking Changes - DotNet 7.md

```diff
@@ -1,5 +1,28 @@
 # This document lists known breaking changes in Roslyn after .NET 6 all the way to .NET 7.
 
+## All locals of restricted types are disallowed in async methods 
```

Comment:
Is it possible for you to include a sample compiler error (and "CS9104") somewhere? It would make this more discoverable to people searching online.

Reply:
Done